spec/validators/immutable_once_set_validator_spec.rb

```diff
@@ -0,0 +1,62 @@
+RSpec.describe ImmutableOnceSetValidator, type: :model do
+  subject { instance.tap(&:valid?).errors[:attribute] }
+
+  let(:instance) { test_class.new(attribute: new_value, attribute_was: current_value) }
+  let(:test_class) do
+    Class.new do
+      include ActiveModel::Model
+      attr_accessor :attribute, :attribute_was
+
+      validates :attribute, immutable_once_set: true
+    end
+  end
+
+  context "when attribute is `nil` and changing to `true`" do
+    let(:current_value) { nil }
+    let(:new_value) { true }
+
+    it { is_expected.to be_empty }
+  end
+
+  context "when attribute is `nil` and changing to `nil`" do
+    let(:current_value) { nil }
+    let(:new_value) { nil }
+
+    it { is_expected.to be_empty }
+  end
+
+  context "when attribute is `\"set\"` and changing to `\"set\"`" do
```

Comment:
Not sure we need double quotations on these 🙂 can we just have one or the other?